plugin-niassa+pinery/src/main/java/ca/on/oicr/gsi/shesmu/niassa/AnalysisState.java

```diff
@@ -63,28 +60,60 @@ public AnalysisState(Pair<Integer, Integer> accessions, List<AnalysisProvenance>
             .collect(Collectors.toCollection(TreeSet::new));
   }
 
-  public boolean compare(
+  /** Check how much this analysis record matches the data provided */
+  public Pair<AnalysisComparison, AnalysisState> compare(
       LongStream workflowAccessions,
       String majorOliveVersion,
-      String fileAccessions,
-      List<? extends LimsKey> limsKeys) {
+      Set<Integer> inputFiles,
+      List<? extends LimsKey> inputLimsKeys) {
+    // If the WF, version or input files doesn't match, bail
     if (workflowAccessions.noneMatch(a -> workflowAccession == a)
```

Comment:
I don't think this is correct behaviour. I think the the olive should filter equivalent workflows, and we should be left with an action for a workflow that should definitely happen as long as the other conditions are met. With this syntax, I think that you couldn't e.g. run Mutect workflow and Strelka workflow on the same input files.

Reply:
wait, are these workflow accessions or workflow *run* accessions?

Reply:
These are workflow accessions, not workflow run accessions. Mutect and Strelka should have different WF accessions and no “previous” WF accessions should be shared between them.